### PR TITLE
PR2: Centrifuges

### PR DESCRIPTION
**Pull request by bam241**
This PR adds the centrifuge's definition.
moved and fixed the corresponding method from `enrich_functions` to `CentrifugeConfig`

this need to be reviewed after #23  (as all #23 commits are also included here...)

Most of the equations used are defined [here](https://github.com/bam241/publications/blob/cvt_cascadeenrich/paper/paper.pdf)

### CODE REVIEW

**Comment by gonuke** [comment]
Not sure why, but this appears to need a rebase against the `cascadeenrich_PR_branch`...

**Comment by bam241** [comment]
this is strange, but it worked

File: src/CentrifugeConfig.cc [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+  // withdrawl radius for heavy isotope
+  // Glaser 2009 says operationally it ranges from 0.96-0.99
+  double r_2 = 0.975 * a;
+
+  // Glaser v_a < 380 >> r_12 = cte = 0.534
+  double r_12 = 0;
+  if (v_a > 380){
+    r_12 = std::sqrt(
+      1.0 - (2.0 * gas_const * temp * (log(x)) / M / (pow(v_a, 2))));
```
Comment (gonuke) [inline src/CentrifugeConfig.cc]: for squaring a variable, simpler just use `v_a * v_a`

File: src/CentrifugeConfig.cc [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  double major_term =
+      0.5 * cut * (1.0 - cut) * (pow(C_therm, 2)) * C_scale *
+      pow(((bracket1 * (1 - exp1)) + (bracket2 * (1 - exp2))), 2);  // kg/s
+  double del_U = feed * major_term * eff;                           // kg/s
+
+  return del_U;
+}
+
+double CentrifugeConfig::CalcCTherm(double v_a, double temp, double dM) {
+  double c_therm = (dM * (pow(v_a, 2))) / (2.0 * gas_const * temp);
```
Comment (gonuke) [inline src/CentrifugeConfig.cc]: `v_a * v_a` is better than `pow(v_a,2)`

File: src/CentrifugeConfig.cc [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  double r12_sq = pow(r_12, 2);
+  double C_scale = (pow((r_2 / a), 4)) * (pow((1 - r12_sq), 2));
+  double bracket1 = (1 + flow_internal) / cut;
+  double exp1 = exp(-1.0 * A_p * Z_p);
+  double bracket2 = flow_internal / (1 - cut);
+  double exp2 = exp(-1.0 * A_w * (height - Z_p));
+
+  // Glaser eqn 10 (Ratz Equation)
+  double major_term =
+      0.5 * cut * (1.0 - cut) * (pow(C_therm, 2)) * C_scale *
```
Comment (nuclearkatie) [inline src/CentrifugeConfig.cc]: `C_therm * C_therm`

File: src/CentrifugeConfig.h [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  friend CascadeConfig;
+
+ public:
+  CentrifugeConfig();
+  CentrifugeConfig(double v_a, double h, double d, double feed, double T,
+                   double eff, double M, double dM, double x, double i_flow);
+
+  double ComputeDeltaU(double cut); // compute the solution of the Raetz equation using all the paramters valus and the provided cut value
+
+  double dM; // molar mass diff between the 2 components of the gaz default 0.003kg/mol (u235/U238)
```
Comment (nuclearkatie) [inline src/CentrifugeConfig.h]: I believe you meant gas?

File: src/CentrifugeConfig.h [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  double M;  // gaz molar mass, default UF6 0.352 kg/mol
```
Comment (nuclearkatie) [inline src/CentrifugeConfig.h]: gas

File: src/CentrifugeConfig.h [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  double x;  // pressure ration -> drive r1/r2 ratio
+  double flow_internal; // countercurrent-to-feed ratios k range between 2 and 4
+
+  double v_a; // peripheral velocities
+  double height; // centrifugre height
+  double diameter; // centrifuge diameter
+  double feed; // feed flow rate
+  double temp; // average gaz temperature
```
Comment (nuclearkatie) [inline src/CentrifugeConfig.h]: gas

**Comment by bam241** [comment]
@gonuke @nuclearkatie just addressed your different comments.

**Pull request merged**